pkg/plugins/discovery/k8s/util/util_test.go

```diff
@@ -129,4 +132,321 @@ var _ = Describe("Util", func() {
 			Expect(copy).ToNot(BeIdenticalTo(original))
 		})
 	})
+
+	Describe("FindPort()", func() {
+		Describe("should return a correct port number", func() {
+			type testCase struct {
+				pod      string
+				svcPort  string
+				expected int
+			}
+
+			DescribeTable("should correctly find a matching port in a given Pod",
+				func(given testCase) {
+					// setup
+					pod := kube_core.Pod{}
+					svcPort := kube_core.ServicePort{}
+
+					// when
+					err := yaml.Unmarshal([]byte(given.pod), &pod)
+					// then
+					Expect(err).ToNot(HaveOccurred())
+
+					// when
+					err = yaml.Unmarshal([]byte(given.svcPort), &svcPort)
+					// then
+					Expect(err).ToNot(HaveOccurred())
+
+					// when
+					actual, err := FindPort(&pod, &svcPort)
+					// then
+					Expect(err).ToNot(HaveOccurred())
+					// and
+					Expect(actual).To(Equal(given.expected))
+				},
+				Entry("Service with `targetPort` as a number (TCP)", testCase{
+					pod: `
+                    spec:
+                      containers:
+                      - name: container-1
+                        ports: [] # notice that actual container ports become irrelevant when Service has 'targetPort' as a number
```

Comment:
Will the service targeting a port that is not specified in the container actually work?
I thought that if you don't specify port, you cannot access it from the outside.

Reply:
Yes, it will work. 

Here is a [quote](https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.17/#container-v1-core) from Kubernetes API Reference about `Container.ports` field:

> List of ports to expose from the container. Exposing a port here gives the system additional information about the network connections a container uses, but is primarily informational. Not specifying a port here DOES NOT prevent that port from being exposed. Any port which is listening on the default "0.0.0.0" address inside a container will be accessible from the network. Cannot be updated.

And here is [exactly the same logic](https://github.com/kubernetes/kubernetes/blob/v1.17.3/pkg/api/v1/pod/util.go#L30-L51) of finding a match between `Pod` and `Service` inside Kubernetes [Endpoint Controller](https://github.com/kubernetes/kubernetes/blob/v1.17.3/pkg/controller/endpoint/endpoints_controller.go#L444-L460).